app/Models/User.php

```diff
@@ -74,4 +77,55 @@ public function getCanAttribute()
     {
         return $this->getAllPermissions()->pluck('name');
     }
+
+    public function getUserNumberOfBookingRequestPerPeriod()
+    {
+        $numberOfBookingRequestPerPeriod = null;
+
+        foreach ($this->roles as $role) {
+            if ($role->number_of_bookings_per_period > $numberOfBookingRequestPerPeriod) {
+                $numberOfBookingRequestPerPeriod = $role->number_of_bookings_per_period;
+            }
+        }
+
+        return $numberOfBookingRequestPerPeriod;
+    }
+
+    public function getUserNumberOfDaysPerPeriod()
+    {
+        $numberOfDaysPerPeriod = null;
+
+        foreach ($this->roles as $role) {
+            if ($role->number_of_days_per_period > $numberOfDaysPerPeriod) {
+                $numberOfDaysPerPeriod = $role->number_of_days_per_period;
+            }
+        }
+
+        return $numberOfDaysPerPeriod;
+    }
+
+  public function canMakeAnotherBookingRequest($startDate): bool
+  {
+    $numberOfBookingRequestPerPeriod = $this->getUserNumberOfBookingRequestPerPeriod();
+    $numberOfDaysPerPeriod = $this->getUserNumberOfDaysPerPeriod();
+
+    if (is_null($numberOfBookingRequestPerPeriod) && is_null($numberOfDaysPerPeriod)) {
+      return true;
+    }
+
+    if (Carbon::parse($startDate)->diffInDays(Carbon::now()) > $numberOfDaysPerPeriod) {
+      return true;
+    }
+
+    $nbOfBookingRequest = BookingRequest::query()
+      ->whereHas('reservations', function (Builder $query) use ($numberOfDaysPerPeriod) {
+        $query->where('start_time', '>', Carbon::now());
+        $query->where('start_time', '<', Carbon::now()->addDays($numberOfDaysPerPeriod));
+      })
+      ->where('user_id', '=', $this->id)
+      ->where('status', 'like', '%approved%')
+      ->count();
+
+    return $nbOfBookingRequest < $numberOfBookingRequestPerPeriod;
```

Comment:
This seems to be returning true no matter what, my guess is request count is always returning 0 for some reason.